src/Raven.Server/Storage/Layout/StorageLoader.cs

```diff
@@ -0,0 +1,138 @@
+                var preRtmJournalPath = basePath.Combine("Journal");
+                var journalsPath = options.JournalPath;
+                if (Directory.Exists(preRtmJournalPath.FullPath))
+                {
+                    TryMoveJournals(preRtmJournalPath, journalsPath);
+                    Directory.Delete(preRtmJournalPath.FullPath);
+                }
+                else
+                {
+                    if (TryMoveJournals(basePath, journalsPath) == false)
+                    {
+                        var message =
+                            $"Failed to open a storage at {options} due to invalid or missing journal files. In order to load the storage successfully we need all journals to be not corrupted. ";
+
+                        string ravenServer = Assembly.GetAssembly(typeof(StorageLoader)).GetName().Name + (PlatformDetails.RunningOnPosix ? string.Empty : ".exe");
+
+                        switch (type)
+                        {
+                            case StorageEnvironmentWithType.StorageEnvironmentType.Index:
+                                message += "Please reset the index in order to recover from this error.";
+                                break;
+                            case StorageEnvironmentWithType.StorageEnvironmentType.Documents:
+                                message +=
+                                    "You can load a database by starting the server in dangerous mode temporary so it will ignore invalid journals on startup but you need to " +
+                                    $"export and import database immediately afterwards: {Environment.NewLine}" +
+                                    $"{ravenServer} --{RavenConfiguration.GetKey(x => x.Storage.IgnoreInvalidJournalErrors)}=true{Environment.NewLine}" +
+                                    "If you won't be able to export the database successfully you need to use Voron.Recovery tool";
+                                
+                                break;
+                            case StorageEnvironmentWithType.StorageEnvironmentType.Configuration:
+                                message += $"You can delete the configuration storage folder at '{basePath.FullPath}' and restart the server. It will be recreated on database startup.";
+                                break;
+                            case StorageEnvironmentWithType.StorageEnvironmentType.System:
+                                message += "You can start the server in dangerous mode temporary so it will ignore invalid journals on startup:" +
+                                           $"{ravenServer} --{RavenConfiguration.GetKey(x => x.Storage.IgnoreInvalidJournalErrors)}=true{Environment.NewLine}" +
+                                           $"Eventually you should delete the storage configuration at '{basePath.FullPath}' and create your databases again with the usage of existing data.";
+                                break;
```

Comment:
Depending on the storage env type I wanted to suggest different actions to recover from this state. Please let me know if you see the potential improvement of those messages.